elastic-mapping-updater/src/test/java/com/github/cafdataprocessing/elastic/tools/test/ElasticMappingUpdaterIT.java

```diff
@@ -810,161 +677,111 @@ public void testRemoveParams() throws Exception
+        final Property cityProp = props.get("city");
+        assertFalse(cityProp.text().fields().isEmpty());
 
-        @SuppressWarnings("unchecked")
-        final Map<String, Object> prop3Mapping = (Map<String, Object>) props.get("message");
-        assertFalse(prop3Mapping.containsKey("ignore_above")); // has been removed
+        final Property messageProp = props.get("message");
+        assertNull(messageProp.keyword().ignoreAbove()); // has been removed
 
-        @SuppressWarnings("unchecked")
-        final Map<String, Object> prop4Mapping = (Map<String, Object>) props.get("number_one");
-        assertFalse(prop4Mapping.containsKey("ignore_malformed")); // has been removed, as expected
+        final Property numberOneProp = props.get("number_one");
+        assertNull(numberOneProp.integer().ignoreMalformed()); // has been removed, as expected
 
-        @SuppressWarnings("unchecked")
-        final Map<String, Object> prop20Mapping = (Map<String, Object>) props.get("dummy_message2");
-        assertTrue(prop20Mapping.containsKey("index_options")); // has not been removed, as expected
+        final Property dummyMessage2Prop = props.get("dummy_message2");
+        assertNotNull(dummyMessage2Prop.text().indexOptions()); // has not been removed, as expected
 
-        @SuppressWarnings("unchecked")
-        final Map<String, Object> prop24Mapping = (Map<String, Object>) props.get("latency");
-        assertTrue(prop24Mapping.containsKey("meta"));
+        final Property latencyProp = props.get("latency");
+        assertFalse(latencyProp.long_().meta().isEmpty());
 
-        @SuppressWarnings("unchecked")
-        final Map<String, Object> prop5Mapping = (Map<String, Object>) props.get("status_code");
-        assertTrue(prop5Mapping.containsKey("null_value")); // not removed, as expected
+        final Property statusCodeProp = props.get("status_code");
+        assertNotNull(statusCodeProp.keyword().nullValue()); // not removed, as expected
 
-        @SuppressWarnings("unchecked")
-        final Map<String, Object> prop27Mapping = (Map<String, Object>) props.get("names2");
-        assertTrue(prop27Mapping.containsKey("position_increment_gap"));
+        final Property names2Mapping = props.get("names2");
+        assertNotNull(names2Mapping.text().positionIncrementGap());
 
-        @SuppressWarnings("unchecked")
-        final Map<String, Object> prop28Mapping = (Map<String, Object>) props.get("manager");
-        assertTrue(prop28Mapping.containsKey("properties"));
+        final Property managerProp = props.get("manager");
+        assertFalse(managerProp.object().properties().isEmpty());
 
-        @SuppressWarnings("unchecked")
-        final Map<String, Object> prop6Mapping = (Map<String, Object>) props.get("title_z");
-        assertTrue(prop6Mapping.containsKey("search_analyzer"));
+        final Property titleZProp = props.get("title_z");
+        assertNull(titleZProp.text().searchAnalyzer());
```

Comment:
This is the only test expectation that has been changed. 

`searchAnalyzer` is in `MODIFIABLE_PROPERTIES` and therefore can be removed if the mapping change is safe.

'template14.json' and 'template15.json' have 'boost' incorrectly written and therefore parsed as 'integer' but would be converted to 'double' in the template.

Now, because of the serialization that happens in `getObjectAsHashMap` before checking mapping changes they are correctly parsed as double. Therefore they are no longer recognized as an unsafe mapping in `isMappingChangeSafe` so the changes to `title_z` are made and `search analyzer` is removed.